tests/smoketest.py

```diff
@@ -43,45 +42,58 @@ def run(command, runner=check_call, **kwargs):
     run("git clone --depth 1 --recurse-submodules https://github.com/HumanCellAtlas/data-store-cli")
 
 run("http --check-status https://${API_HOST}/v1/swagger.json > data-store-cli/swagger.json")
-run("pip install -r data-store-cli/requirements.txt")
-run("python -c 'import sys, hca.dss.regenerate_api as r; r.generate_python_bindings(sys.argv[1])' swagger.json",
-    cwd="data-store-cli")
-run("find data-store-cli/hca -name '*.pyc' -delete")
-run("pip install --upgrade --no-deps .", cwd="data-store-cli")
+venv = tempfile.mkdtemp()
+try:
+    run(f"virtualenv {venv}")
+    venv_bin = os.path.join(venv, 'bin', '')
+    run(f"{venv_bin}pip install -r data-store-cli/requirements.txt")
+    run(f"{venv_bin}python -c 'import sys, hca.dss.regenerate_api as r; "
+        f"r.generate_python_bindings(sys.argv[1])' swagger.json",
+        cwd="data-store-cli")
+    run("find data-store-cli/hca -name '*.pyc' -delete")
+    run(f"{venv_bin}pip install --upgrade --no-deps .", cwd="data-store-cli")
 
-sample_id = str(uuid.uuid4())
-bundle_dir = "data-bundle-examples/10X_v2/pbmc8k"
-with open(os.path.join(bundle_dir, "async_copied_file"), "wb") as fh:
-    fh.write(os.urandom(ASYNC_COPY_THRESHOLD + 1))
+    sample_id = str(uuid.uuid4())
+    bundle_dir = "data-bundle-examples/10X_v2/pbmc8k"
+    with open(os.path.join(bundle_dir, "async_copied_file"), "wb") as fh:
+        fh.write(os.urandom(ASYNC_COPY_THRESHOLD + 1))
 
-run(f"cat {bundle_dir}/sample.json | jq .uuid=env.sample_id | sponge {bundle_dir}/sample.json",
-    env=dict(os.environ, sample_id=sample_id))
-run(f"hca dss upload --replica aws --staging-bucket $DSS_S3_BUCKET_TEST --file-or-dir {bundle_dir} > upload.json")
-run("hca dss download --replica aws $(jq -r .bundle_uuid upload.json)")
-for i in range(10):
-    try:
-        run("http -v --check-status https://${API_HOST}/v1/bundles/$(jq -r .bundle_uuid upload.json)?replica=gcp")
-        break
-    except SystemExit:
-        time.sleep(1)
-else:
-    parser.exit(RED("Failed to replicate bundle from AWS to GCP"))
-run("hca dss download --replica gcp $(jq -r .bundle_uuid upload.json)")
+    run(f"cat {bundle_dir}/sample.json | jq .uuid=env.sample_id | sponge {bundle_dir}/sample.json",
+        env=dict(os.environ, sample_id=sample_id))
+    run(f"{venv_bin}hca dss upload "
```

Comment:
Does executing {venv_bin}/hca use that virtualenv even if one doesn't activate that venv?

Reply:
Yes. Running a binary from the venv's bin activates that venv on the fly for the duration of the process.